### PR TITLE
Create/Update and Verify Service connection using configuration file

### PR DESCRIPTION
**Pull request by ganeshg2491**
- Create or Update Service connection using configuration file
- Verify Service connection
- readme Documentation added

### CODE REVIEW

File: infra/config/service-connections/tier2-service-connection.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "azureRMServiceConnections": [
+        {
+            "displayName": "#{{ ?? }}",
```
Comment (ganeshg2491) [inline infra/config/service-connections/tier2-service-connection.json:4]: I will replace the placholders ?? with the actual variables

File: .azuredevops/deploy-platform-bootstrap-serviceconnections.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ resources:
       name: DEFRA/adp-pipeline-common
       endpoint: DEFRA
       type: github
-      ref: main 
+      ref: gg/214939-sc-accesstoken #main
```
Comment (ganeshg2491) [inline .azuredevops/deploy-platform-bootstrap-serviceconnections.yaml:15]: Update this to main when dependent [PR ](https://dev.azure.com/defragovuk/DEFRA-DEVOPS-COMMON/_git/Defra.Pipeline.Common/pullrequest/36783) is completed and merged to main

File: scripts/service-connection/utilities/Get-DefaultHeadersWithAccessToken.ps1
Comment (ppotturi) [inline scripts/service-connection/utilities/Get-DefaultHeadersWithAccessToken.ps1:1]: @ganeshg2491 I think we want to create a module for all utilities

Comment (ganeshg2491) [reply scripts/service-connection/utilities/Get-DefaultHeadersWithAccessToken.ps1:1]: @ppotturi : Added module

**Review by DaRoszko — APPROVED** [review]

**Review by DaRoszko — APPROVED** [review]

**Pull request merged**